### PR TITLE
Added constructors taking PyObject

### PR DESCRIPTION
**Pull request by prasadsom**
Added constructors taking `PyObject`

#### Description
Added overloaded constructor with `PyObject` as parameter. `PeriodCountConsolidatorBase` constructor convert `PyObject` to delegate .

#### Related Issue
#3780 
#### Motivation and Context
Though there is workaround available, suggestion is not to rely on `System` package. So overloaded C# class constructors

#### Requires Documentation Change
NA

#### How Has This Been Tested?
Using snippet in #3780 

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Common/Data/Consolidators/PeriodCountConsolidatorBase.cs [hunk 1]
```diff
@@ -89,6 +89,17 @@ protected PeriodCountConsolidatorBase(Func<DateTime, CalendarInfo> func)
             _period = Time.OneSecond;
         }
 
+        /// <summary>
+        /// Creates a consolidator to produce a new <typeparamref name="TConsolidated"/> instance representing the last count pieces of data or the period, whichever comes first
+        /// </summary>
+        /// <param name="funcpyobj">Python function object that defines the start time of a consolidated data</param>
+        protected PeriodCountConsolidatorBase(PyObject funcpyobj)
+        {
```
Comment (AlexCatarino) [inline Common/Data/Consolidators/PeriodCountConsolidatorBase.cs]: Why not just:
```csharp
        protected PeriodCountConsolidatorBase(PyObject funcpyobj)
            : this(funcpyobj.ConvertToDelegate<Func<DateTime, CalendarInfo>>())
        {
        }
```

Comment (prasadsom) [reply Common/Data/Consolidators/PeriodCountConsolidatorBase.cs]: Modified it now . Added overloaded constructors to derived classes from `PeriodCountConsolidatorBase`.
Still not added unit test. Was trying to find correct place to do that.

Comment (prasadsom) [reply Common/Data/Consolidators/PeriodCountConsolidatorBase.cs]: Added unit tests.

**Comment by prasadsom** [comment]
Removed code where, I tried to convert C# delegate to PyObject. it's failing on CI unit test run, though it's alright on my win/linux dev run. Just kept step where function from Python module is converted to PyObject. 
There is a separate Python file, which can be tested in isolation.

**Comment by jaredbroad** [comment]
Superceeded/extended by #3816

**Pull request closed without merging**